### PR TITLE
newsflash: 2.3.1 -> 3.0.2

### LINKED ISSUE #258497: Update request: newsflash 2.3.1 → 3.0.1
- Package name: newsflash
- Latest released version: 3.0.1

- Current version on the unstable channel: 2.3.1
- Current version on the stable/release channel: 2.3.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@kira-bruneau 
@Stunkymonkey 

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

~~currently blocked by #247766~~

fixes #258497

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/feedreaders/newsflash/default.nix [hunk 1]
```diff
@@ -24,22 +24,24 @@
...
 
   src = fetchFromGitLab {
     owner = "news-flash";
     repo = "news_flash_gtk";
     rev = "refs/tags/v.${finalAttrs.version}";
-    sha256 = "sha256-JUAlDc2mp8M0vjiWcDoyBw/sKCmd4J8e9wEwZoiW0AE=";
+    sha256 = "sha256-tJKr2bGkdpEb+25eN0ZfHhEDl5Zdf8fdaC/rNMbH8Ws=";
   };
 
+  cargoHash = "sha256-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=";
```
Comment (kira-bruneau) [inline pkgs/applications/networking/feedreaders/newsflash/default.nix:36]: We shouldn't need this `cargoHash` using `cargoDeps` and the lock file, but other than that - this looks good to merge once #227588 is in!

Comment (Stunkymonkey) [reply pkgs/applications/networking/feedreaders/newsflash/default.nix]: removed

**Comment by Stunkymonkey** [comment]
@kira-bruneau : I added the `blueprint-compiler` and rebased, because the GNOME PR is merged.

**Comment by Stunkymonkey** [comment]
it starts and seems to work fine, but it prints:
```
23:20:55 - ERROR - Failed to set background message: Portal request failed: org.freedesktop.portal.Error.NotAllowed: Only sandboxed applications can set background status (news_flash_gtk::app:2501)
```
no idea about it.

**Comment by surfaceflinger** [comment]
> it starts and seems to work fine, but it prints:
> 
> ```
> 23:20:55 - ERROR - Failed to set background message: Portal request failed: org.freedesktop.portal.Error.NotAllowed: Only sandboxed applications can set background status (news_flash_gtk::app:2501)
> ```
> 
> no idea about it.

My guess is that's coming from support for that [GNOME 44 feature](https://linuxiac.com/gnome-background-apps/) which well... works only with Flatpak and maybe snap(?). We can probably ignore it or patch it out so it doesn't attempt to use that api.

**Comment by surfaceflinger** [comment]
btw second commit should probably be `newsflash: adjust binary name`

**Comment by Stunkymonkey** [comment]
@surfaceflinger i thought about squashing the commits before merging.

**Comment by Stunkymonkey** [comment]
so i did use this for 3 days. Works perfectly. I think this is ready for merge.

Not sure if we should patch out the "gnome-background" stuff. gnome users might want to rely on it. We better should report upstream, that there are users without gnome.

**Comment by surfaceflinger** [comment]
> so i did use this for 3 days. Works perfectly. I think this is ready for merge.
> 
> Not sure if we should patch out the "gnome-background" stuff. gnome users might want to rely on it. We better should report upstream, that there are users without gnome.

not sure what u meant by `gnome users might want to rely on it`, i meant there's no need to touch it (in newsflash, not gnome, i assume it would work if you installed newsflash from flatpak, not nix) unless this error in log makes u mad.

basically newsflash tries to use that new gnome api that (currently?) is available only for apps from flatpak. it can just be ignored :p

**Comment by surfaceflinger** [comment]
anyways, i'm using this too right now, imported my opml and so far works great 🙂

**Comment by kira-bruneau** [comment]
Hmm... I wonder if there's some way we can workaround around that with the way we package newsflash - but I think this is good to merge without it since it's non-breaking.

Thanks @Stunkymonkey!!

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 258567` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>newsflash</li>
  </ul>
</details>

**Comment by huantianad** [comment]
Note t hat I also get
```
ERROR - Failed to set background message: Portal request failed: org.freedesktop.portal.Error.NotAllowed: Only sandboxed applications can set background status (news_flash_gtk::app:2461)
```
when running the old version of Newsflash on KDE Plasma. The message appears each time I toggle the "Keep running; fetch updates in the background" option in settings as well.

It would be nice to get background support working, I'd assume that's a modern replacement for the systray and notification stack? But I feel that would require a decent amount of work.

PS, I assume adding a desktop file to `~/.config/autostart` with `io.gitlab.news_flash.NewsFlash --headless ` is the main way to auto start the app when running with background on?

**Pull request merged**